storage/src/test/java/org/apache/kafka/server/log/remote/storage/LocalTieredStorageTest.java

```diff
@@ -0,0 +1,677 @@
+         *
+         * @param id The unique ID of the remote log segment and associated resources (e.g. offset and time indexes).
+         * @param expected The expected content.
+         */
+        public void verifyFetchedTransactionIndex(final RemoteLogSegmentId id, final byte[] expected) {
+            verifyFileContents(metadata -> remoteStorage.fetchIndex(metadata, TRANSACTION), id, expected);
+        }
+
+        /**
+         * Verifies the content of the remote leader epoch checkpoint matches with the {@code expected} array.
+         *
+         * @param id The unique ID of the remote log segment and associated resources (e.g. offset and time indexes).
+         * @param expected The expected content.
+         */
+        public void verifyLeaderEpochCheckpoint(final RemoteLogSegmentId id, final byte[] expected) {
+            verifyFileContents(metadata -> remoteStorage.fetchIndex(metadata, LEADER_EPOCH), id, expected);
+        }
+
+        /**
+         * Verifies the content of the remote producer snapshot matches with the {@code expected} array.
+         *
+         * @param id The unique ID of the remote log segment and associated resources (e.g. offset and time indexes).
+         * @param expected The expected content.
+         */
+        public void verifyProducerSnapshot(final RemoteLogSegmentId id, final byte[] expected) {
+            verifyFileContents(metadata -> remoteStorage.fetchIndex(metadata, PRODUCER_SNAPSHOT), id, expected);
+        }
+
+        private void verifyFileContents(final Function<RemoteLogSegmentMetadata, InputStream> actual,
+                                        final RemoteLogSegmentId id,
+                                        final byte[] expected) {
+            try {
+                final InputStream in = actual.apply(newMetadata(id));
+                assertArrayEquals(expected, readFully(in));
+            } catch (RemoteStorageException | IOException e) {
+                throw new AssertionError(e);
+            }
+        }
+
+        private RemoteLogSegmentMetadata newMetadata(final RemoteLogSegmentId id) {
+            return new RemoteLogSegmentMetadata(id, 0, 0, -1L, -1, 1000,
+                    1024, Collections.singletonMap(0, 0L));
+        }
+
+        private String getStorageRootDirectory() {
+            try {
+                return remoteStorage.getStorageDirectoryRoot();
+
+            } catch (RemoteStorageException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        private void assertFileExists(final Path path) {
+            if (!path.toFile().exists()) {
+                throw new AssertionError(format("File %s does not exist", path));
+            }
+        }
+
+        private void assertFileDoesNotExist(final Path path) {
+            if (path.toFile().exists()) {
+                throw new AssertionError(format("File %s should not exist", path));
```

Comment:
nit: `assertFalse(path.toFile().exists(), format("File %s does not exist", path));`